daemon/graphdriver/zfs/zfs.go

```diff
@@ -361,6 +361,11 @@ func (d *Driver) Remove(id string) error {
 	name := d.zfsPath(id)
 	dataset := zfs.Dataset{Name: name}
 	err := dataset.Destroy(zfs.DestroyRecursive)
+	if err != nil && strings.Contains(fmt.Sprint(err), "dataset does not exist") {
```

Comment:
you may probably do a stricter check.

```go
errZfs, ok := err.(zfs.Error)
if ok && errZfs.Stderr == "dataset does not exist" {
   ...
}
```

or go with `err.Error()` instead of `fmt.Sprint(err)`, see https://pkg.go.dev/builtin@go1.23.1#error

Happy Go hacking mate!

Reply:
Done. Unfortunately, the stderr field contains more the whole output of the zfs command
```text
Stderr: "cannot open 'rpool/zzztest/doesnotexist': dataset does not exist\n",
```